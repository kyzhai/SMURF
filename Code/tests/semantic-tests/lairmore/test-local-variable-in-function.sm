f :: Int-> Int
f x = x
y = f 1
main = []
