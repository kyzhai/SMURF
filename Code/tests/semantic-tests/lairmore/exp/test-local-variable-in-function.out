Int and Int
Program: 
	SFuncdec: 
		f (x) :: Int -> Int = x
	New Scope: 
		ID: x  :: Int

	
	SVardef: 
		ID: y  :: Int

	f 1
	SMain: []
Global Scope: 
	ID: main  :: []
	ID: f x :: Int -> Int
	ID: y  :: Int
	ID: print x :: a -> a
	ID: random  :: Int
	ID: head h : t :: [a] -> a
	ID: tail h : t :: [a] -> [a]

